applications/GeoMechanicsApplication/custom_utilities/convergence_criteria_factory.hpp

```diff
@@ -26,25 +28,32 @@ class ConvergenceCriteriaFactory
     static std::shared_ptr<ConvergenceCriteriaType> Create(const Parameters& rSolverSettings)
     {
         KRATOS_ERROR_IF_NOT(rSolverSettings.Has("convergence_criterion"))
-        << "No convergence_criterion is defined, aborting.";
+            << "No convergence_criterion is defined, aborting.";
 
-        if (rSolverSettings["convergence_criterion"].GetString() == "displacement_criterion")
+        if (rSolverSettings["convergence_criterion"].GetString() ==
+            "displacement_criterion")
         {
-            Parameters convergenceInputs;
-
-            std::vector<std::string> entries_to_copy = {"displacement_absolute_tolerance", "displacement_relative_tolerance"};
-            for (const std::string& entry : entries_to_copy)
-            {
-                if (rSolverSettings.Has(entry))
-                {
-                    convergenceInputs.AddValue(entry, rSolverSettings[entry]);
-                }
-            }
-            return std::make_shared<DisplacementCriteria<TSparseSpace, TDenseSpace>>(convergenceInputs);
+            const std::vector<std::string> entries_to_copy = {
+                "displacement_absolute_tolerance",
+                "displacement_relative_tolerance"};
+            Parameters convergence_inputs = ParametersUtilities::ExtractParameters(rSolverSettings, entries_to_copy);
+            return std::make_shared<DisplacementCriteria<TSparseSpace, TDenseSpace>>(convergence_inputs);
+        }
+        if (rSolverSettings["convergence_criterion"].GetString() ==
+            "residual_criterion")
+        {
+            const std::vector<std::string> entries_to_copy = {
+                "residual_absolute_tolerance", "residual_relative_tolerance"};
+            Parameters convergence_inputs = ParametersUtilities::ExtractParameters(rSolverSettings, entries_to_copy);
```

Comment:
And the same thing here:
```suggestion
            const auto convergence_inputs = ParametersUtilities::ExtractParameters(rSolverSettings, entries_to_copy);
```